### DISCUSSION TITLE
Survival modeling

### DISCUSSION THREAD

**Discussion by chanansh** [catboost/catboost#2730] [Q&A]
https://github.com/catboost/tutorials/blob/master/regression/survival.ipynb - talks about censoring but when one trains the model the model does not get as input whether the duration is censored or not

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
in SurvivalAft models, target is 2-dimensional, [observation_start, observation_end]
in the tutorial, observation_start == y_lower, observation_end == y_upper, and -1 means 'infinity'

**Comment by chanansh** [comment]
How does this consistent with the following line written in the notebook?:
```python
target['label'] = np.where(target['death'], target['futime'], - target['futime'])
```

**Reply by CollinSakal** [reply]
For using Cox loss the [documentation](https://catboost.ai/docs/en/concepts/loss-functions-regression#Cox) states that the label (time to event or censoring) should be >0 for observations where an event was observed and <0 for observations where censoring occurred. That formulation seems to allow the model to account for censoring internally using a singular input.
